src/backups.py

```diff
@@ -326,7 +418,7 @@ def _on_create_backup_action(self, event) -> None:
                 # on the replicas (that happens when TLS is not enabled).
                 command.append("--no-backup-standby")
             stdout, stderr = self._execute_command(command)
-            backup_id = self._list_backups_ids(show_failed=True)[-1]
+            backup_id = list(self._list_backups(show_failed=True).keys())[-1]
```

Comment:
`.keys()` returns keys in random order, we need or sort it or use OrderedDict in `_list_backups`

Reply:
Fixed on https://github.com/canonical/postgresql-k8s-operator/pull/147/commits/177f19023274bd3ea2659f1aa664293b568200f4 by using `OrderedDict`.